### PR TITLE
test: page-level request/response events with SW fetch handlers and interception

### PR DESCRIPTION
**Pull request by rwoll**
Repro for #15474 derived from use-case by @Meir017!

### CODE REVIEW

File: tests/library/chromium/chromium.spec.ts [hunk 1]
```diff
@@ -250,6 +250,99 @@ test('setOffline', async ({ context, page, server }) => {
...
+  ]);
+
+  await Promise.all([
+    context.waitForEvent('request', request => !!request.serviceWorker()),
+    sw.evaluate(async () => {
+      const cache = await caches.open('v1');
+      await cache.add('scripts/resource-1.js');
+    })
+  ]);
+  console.info('wait for request from window intercepted by service-worker');
```
Comment (dgozman) [inline tests/library/chromium/chromium.spec.ts:329]: Drop this one.

File: tests/library/chromium/chromium.spec.ts [hunk 2]
```diff
@@ -250,6 +250,99 @@ test('setOffline', async ({ context, page, server }) => {
...
+  });
+  const [ sw ] = await Promise.all([
+    context.waitForEvent('serviceworker'),
+    page.goto(server.PREFIX + '/test-page.html'),
+    page.waitForFunction('activationPromise'),
+  ]);
+
+  await Promise.all([
+    context.waitForEvent('request', request => !!request.serviceWorker()),
+    sw.evaluate(async () => {
```
Comment (dgozman) [inline tests/library/chromium/chromium.spec.ts:324]: Perhaps, just reloading the page would trigger the cache response?

**Review by dgozman — APPROVED** [review]

File: tests/library/chromium/chromium.spec.ts [hunk 3]
```diff
@@ -250,6 +250,99 @@ test('setOffline', async ({ context, page, server }) => {
...
+      const cache = await caches.open('v1');
+      await cache.add('scripts/resource-1.js');
+    })
+  ]);
+  console.info('wait for request from window intercepted by service-worker');
+  await Promise.all([
+    // currently times out here
+    page.waitForRequest('**/scripts/resource-1.js'),
```
[the remark this answers is not in the document]
Comment (rwoll) [reply tests/library/chromium/chromium.spec.ts:332]: I marked with `fixme` so we don't actually run this.

File: tests/library/chromium/chromium.spec.ts (hunk 2 above)
Comment (rwoll) [reply tests/library/chromium/chromium.spec.ts:324]: Good point—let me see if it repros like that instead.

Comment (rwoll) [reply tests/library/chromium/chromium.spec.ts:324]: Oh—we don't load this script on page-load, so this is the first time we're requesting it.

**Pull request merged**